### ISSUE TITLE
Idempotent --install

### ISSUE THREAD

**Issue by epigramengineer** [knadh/listmonk#247]
I'm having a tough time coming up with a declarative deployment strategy for listmonk given that the --install flag is not idempotent. I can see a few ways to fix this:

1. When --install option is provided (or anothor flag --idempotent_install), only run it if the database is clean.
2. Use an exit code other than 0 if the database isn't setup. This is extremely easy and would allow for scripting of `./listmonk || (yes | ./listmonk --install) && ./listmonk`.

**Comment by mr-karan** [comment]
Adding my $0.02

> with a declarative deployment strategy for listmonk

You're confusing the initial step of migration/seeding the DB with actual deployment (running of the process). `listmonk` can be deployed declaratively if you keep these 2 steps isolated from each other. This is how it could be done in K8s, but the broad concepts remain the same, if you're using anything else:

1) Run a one time "Job" with the `--install` flag. At this point, only your DB should be up and running. Even if the app container is running it doesn't matter, the container would fail to start because of the missing tables when it connects to DB. Yes, not quite ideal but I guess it does the job.

2) Once the above Job is done, the container on the next boot will automatically be up and running, because the tables are seeded.

The first step is only manual here, next time if you run even with an `--upgrade`, the migrations being idempotent already, will be skipped.

Although:

> Use an exit code other than 0 if the database isn't setup

I like this approach as well, in case someone doesn't want the hassle of doing Step 1), a different specific exit code which is documented could be a nice hack too.

Will wait for @knadh 's suggestions.

**Comment by knadh** [comment]
Yeah, idempotent DB setups (first time) aren't very common, it would be nice to have them. There can be `--install` (which runs non-destructive "soft" install that only creates the missing DB schema) and an `--install --force` that follows the current behaviour of recreating everything. Let me give this some thought.

In the meanwhile, like @mr-karan suggested, to detect if listmonk is setup you can run `listmonk --upgrade --yes` as a check. If it returns a non 0 response, it means the DB isn't setup and you need to run `--install --yes` (`yes` suppresses the confirmation prompt).

**Comment by knadh** [comment]
https://github.com/knadh/listmonk/commit/3847c67087a80d9c8a25b0b461788bfbde79772b adds an `--idempotent`flag.

eg: `./listmonk --install --yes --idempotent` can be run as many times while the database setup will run only the first time.
